### PR TITLE
Allowing draft moves, PPMs, and orders to be cancelled

### PR DESCRIPTION
**Pull request by breanneboland**
## Description

This changes the Cancel() functions in the move, PPM, and orders models to allow DRAFT status objects to be canceled. (Now, it checks to see if the status is draft or submitted on each; if so, cancellation is possible. If not, errrrrror.) 

## Reviewer Notes

I just omitted the step in the connected tests that once tried to cancel a draft-state move, PPM, or order. Do you feel anything should be added elsewhere to keep the tests as robust as we like, or are we good? 

## Code Review Verification Steps

* [ ] All tests pass.
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/blob/master/docs/backend.md#logging)
* [ ] Request review from a member of a different team.
* [ ] (TRIAL) Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/158612518) for this change

### CODE REVIEW

File: pkg/models/move.go [hunk 1]
```diff
@@ -107,7 +107,7 @@ func (m *Move) Submit() error {
 
 // Cancel cancels the Move and its associated PPMs
 func (m *Move) Cancel(reason string) error {
-	if m.Status != MoveStatusSUBMITTED {
+	if m.Status != MoveStatusSUBMITTED && m.Status != MoveStatusDRAFT {
```
Comment (kilbergr) [inline pkg/models/move.go]: @ntwyman said he couldn't think of a status from which move couldn't be canceled except MoveStatusCOMPLETE--so I actually think it'd be better to do a if m.Status == Complete here.

File: pkg/models/order.go [hunk 2]
```diff
@@ -105,7 +105,7 @@ func (o *Order) Submit() error {
 
 // Cancel cancels the Order
 func (o *Order) Cancel() error {
-	if o.Status != OrderStatusSUBMITTED {
+	if o.Status != OrderStatusSUBMITTED && o.Status != OrderStatusDRAFT {
```
Comment (kilbergr) [inline pkg/models/order.go]: ditto

File: pkg/models/personally_procured_move.go [hunk 3]
```diff
@@ -88,7 +88,7 @@ func (p *PersonallyProcuredMove) ValidateUpdate(tx *pop.Connection) (*validate.E
 
 // Cancel cancels the PPM
 func (p *PersonallyProcuredMove) Cancel() error {
-	if p.Status != PPMStatusSUBMITTED {
+	if p.Status != PPMStatusSUBMITTED && p.Status != PPMStatusDRAFT {
```
Comment (kilbergr) [inline pkg/models/personally_procured_move.go]: ditto

**Comment by breanneboland** [comment]
Current version: 

- You can't cancel a canceled or completed move. 
- You can't cancel a canceled or completed PPM (also, there's now a PPM completed state that, like the move completed state, is not currently manipulated by any of the functions within the model file). 
- You can't cancel a canceled order. 

Pushing this version so that other advisors being busy doesn't hang this up forever.

**Comment by chrisgilmerproj** [comment]
The changes look alright by me but you may need @kilbergr to weigh in on this still.

**Comment by breanneboland** [comment]
I talked to @ntwyman, and he said this approach sounded right. To answer some of the questions that came up earlier today...

**Pull request merged**